web/src/reload.js

```diff
@@ -81,15 +81,15 @@ function preserve_state(send_after_reload, save_pointer, save_narrow, save_compo
 
     if (save_narrow) {
         const $row = message_lists.home.selected_row();
-        if (!narrow_state.active()) {
+        if (narrow_state.filter() !== undefined) {
             if ($row.length > 0) {
                 url += "+offset=" + $row.get_offset_to_window().top;
             }
```

Comment:
This migration is I think inverted; the block in this `if` clause if for `message_lists.home`.

Might be better to just rewrite this without reference to `narrow_state` at all, just accessing the `message_lists` module; that feels more readable. Can be its own little prep commit since it's a bit different from the rest.

Reply:
Makes sense, thanks for catching!